test/server/cards/01_SOR/units/PiratedStarfighter.spec.ts

```diff
@@ -0,0 +1,43 @@
+describe('Pirated Starfighter', function () {
+    integration(function (contextRef) {
+        describe('Pirated Starfighter\'s ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['pirated-starfighter'],
+                    }
+                });
+            });
+
+            it('should not prompt player if no units are available to return to hand', function () {
```

Comment:
nit: just a readability improvement, if a setupTest() call is used only for one test case then we can move the setup into the test case itself and consolidate down to one describe block. This is a pattern we've only recently started doing, you can see the Reinforcement Walker tests as an example.